rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -187,15 +217,63 @@ impl Configurable<StorageConfig> for AdmissionControlledS3Storage {
     async fn try_from_config(config: &StorageConfig) -> Result<Self, Box<dyn ChromaError>> {
         match &config {
             StorageConfig::AdmissionControlledS3(nacconfig) => {
-                let s3_storage = S3Storage::try_from_config(&StorageConfig::S3(
-                    nacconfig.s3_config.clone(),
-                ))
-                .await?;
-                return Ok(Self::new(s3_storage));
+                let s3_storage =
+                    S3Storage::try_from_config(&StorageConfig::S3(nacconfig.s3_config.clone()))
+                        .await?;
+                let policy =
+                    RateLimitPolicy::try_from_config(&nacconfig.rate_limiting_policy).await?;
+                return Ok(Self::new(s3_storage, policy));
             }
             _ => {
                 return Err(Box::new(StorageConfigError::InvalidStorageConfig));
             }
         }
     }
 }
+
+// Prefer enum dispatch over dyn since there could
+// only be a handful of these policies.
+#[derive(Debug)]
+enum RateLimitPolicy {
```

Comment:
To expand on above comment - I think there should be an impl RateLimitPolicy block that is the abstraction for a policy but again, probably fine for now